app/utils/service_client/middleware/request_id.rb

```diff
@@ -0,0 +1,11 @@
+module ServiceClient
+  module Middleware
+    class RequestID < MiddlewareBase
+      def enter(ctx)
+        headers = ctx.fetch(:req).fetch(:headers)
```

Comment:
Right, makes sense. I had an incorrect assumption that fetch wouldn't raise but instead returns a nil when the key is not defined (that's how get works in Clojure). But the fact that it raises when no default is given changes things.